velox/functions/prestosql/aggregates/SumDataSizeForStatsAggregate.cpp

```diff
@@ -0,0 +1,216 @@
+    return sizeof(int64_t);
+  }
+
+  void extractValues(char** groups, int32_t numGroups, VectorPtr* result)
+      override {
+    BaseAggregate::doExtractValues(groups, numGroups, result, [&](char* group) {
+      return *BaseAggregate::Aggregate::template value<int64_t>(group);
+    });
+  }
+
+  void initializeNewGroups(
+      char** groups,
+      folly::Range<const vector_size_t*> indices) override {
+    exec::Aggregate::setAllNulls(groups, indices);
+    for (auto index : indices) {
+      *BaseAggregate ::value<int64_t>(groups[index]) = 0;
+    }
+  }
+
+  void extractAccumulators(char** groups, int32_t numGroups, VectorPtr* result)
+      override {
+    // Partial and final aggregations are the same.
+    extractValues(groups, numGroups, result);
+  }
+
+  void addIntermediateResults(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool mayPushdown) override {
+    BaseAggregate::template updateGroups<true, int64_t>(
+        groups,
+        rows,
+        args[0],
+        [](int64_t& result, int64_t value) {
+          result = checkedPlus(result, value);
+        },
+        mayPushdown);
+  }
+
+  void addSingleGroupIntermediateResults(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool mayPushdown) override {
+    BaseAggregate::updateOneGroup(
+        group,
+        rows,
+        args[0],
+        [](int64_t& result, int64_t value) {
+          result = checkedPlus(result, value);
+        },
+        [](int64_t& result, int64_t value, int /* unused */) {
+          result = checkedPlus(result, value);
+        },
+        mayPushdown,
+        (int64_t)0);
+  }
+
+  void addSingleGroupRawInput(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    doUpdateSingleGroup(group, rows, args[0]);
+  }
+
+  void addRawInput(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    doUpdate(groups, rows, args[0]);
+  }
+
+ protected:
+  void
+  updateOneAccumulator(char* const group, vector_size_t row, int64_t rowSize) {
+    if (decoded_.isNullAt(row)) {
+      return;
+    }
+
+    // Clear null.
+    clearNull(group);
+    // Set sum(current, this).
+    int64_t& current = *value<int64_t>(group);
+    current = checkedPlus(current, rowSize);
+  }
+
+  void
+  doUpdate(char** groups, const SelectivityVector& rows, const VectorPtr& arg) {
+    decoded_.decode(*arg, rows, true);
+    estimateSerializedSizes(arg, rows);
+    vector_size_t sizeIndex = 0;
+    rows.applyToSelected([&](auto row) {
+      updateOneAccumulator(groups[row], row, rowSizes_[sizeIndex++]);
+    });
+  }
+
+  // Estimates the sizes of first numToProcess selected elements in vector.
+  void estimateSerializedSizes(
```

Comment:
Is this a copy-paste from velox/functions/prestosql/aggregates/MaxSizeForStatsAggregate.cpp ? It would be nice to refactor to avoid code duplication.

CC: @xiaoxmeng